configuration.md

```diff
@@ -0,0 +1,433 @@
+associated with this listener. The first Graphite listener is configured during
+[initial installation](installation.md).
+* `check_uuid` is the identifier for all metrics ingested via this listener.
+* `check_name` is a meaningful name that is used in
+  [namespacing](graphite-ingestion.md#namespacing).
+* `account_id` is also part of namespacing, for disambiguation.
+
+#### CLI listener
+
+```
+<listener address="127.0.0.1" port="32322" type="mtev_console">
+  <config>
+    <line_protocol>telnet</line_protocol>
+  </config>
+</listener>
+```
+
+The CLI listener provides a local [telnet
+console](http://circonus-labs.github.io/libmtev/operations/telnet_console.html)
+for interacting with libmtev subsystems, including modifying configuration. As
+there is no authentication mechanism available for this listener, it is
+recommended that it only be operated on the localhost interface.
+
+##### CLI listener address
+
+The IP address on which to listen, or the special `*` to listen on any local IP
+address.
+
+Default: 127.0.0.1
+
+##### CLI listener port
+
+The TCP port number to listen on.
+
+Default: 32322
+
+##### CLI listener type
+
+The CLI listener uses the built-in libmtev type "mtev_console" to allow access
+to the telnet console.
+
+Default: mtev_console
+
+### pools
+
+```
+<pools>
+  <nnt_put concurrency="16"/>
+  <raw_writer concurrency="4"/>
+</pools>
+```
+
+Resource pools within IRONdb are used for various functions, such as reading
+and writing metric data. Some aspects of pool behavior are configurable,
+typically to adjust the amount of concurrency within a given pool.
+
+The defaults presented are widely applicable to most workloads, but may be
+adjusted to improve throughput. Use caution when raising these values too high,
+as it could produce thrashing and _decrease_ performance.
+
+#### pools nnt_put concurrency
+
+The number of job queues used for writing to numeric rollup files. Writes to a
+given rollup file will always occur in the same queue.
+
+Default: the number of physical CPU cores present during installation
+
+#### pools raw_writer concurrency
+
+The number of job queues used for writing to the raw metrics database.
+Additionally, IRONdb will use 4x this number of job queues for reading from the
+raw metrics database.
+
+Default: 4
+
+### rollups
+
+```
+<rollups>
+  <rollup period="60" directory="/irondb/data/{node}/1m"/>
+  <rollup period="300" directory="/irondb/data/{node}/5m"/>
+  <rollup period="1800" directory="/irondb/data/{node}/30m"/>
+  <rollup period="10800" directory="/irondb/data/{node}/3h"/>
+</rollups>
+```
+
+Numeric rollups are produced from the raw metrics database. The schedule for
+when to produce these rollups is controlled in the [raw database
+configuration](#rawdatabase).
+
+Each desired rollup is configured with a `period`, in seconds, and a named
+`directory` for where to store that rollup's files. There will be one file per
+unique metric name, per rollup period.
+
```

Comment:
"How to set these involves a tradeoff between:

- Disk space
- Rollup execution time
- Granularity when fetching rolled data

In general the `raw_database` will cover the most recent time period with very good granularity, if you do not require this much resolution when viewing historic data you can eliminate the finer grained rollups from your configuration"